README.md
<p align="left">
  <a href="https://www.gatsbyjs.org">
    <img alt="Jim Nieters" src="https://www.jimnieters.online/images/Jim-pic.png" width="200" />
  </a>
</p>
<h2 align="left">
 Jim Nieters
</h2>

Jim is an UX leader who loves to inspire and be inspired by design. He believes great design disrupts markets and delights users.

_You can to get in touch with [**Jim Nieters**](https://www.jimnieters.online) or [**Devanand**](mailto:devanand.b@outlook.com)._


[![Netlify Status](https://api.netlify.com/api/v1/badges/11c20974-54ff-4385-a247-377762be7114/deploy-status)](https://app.netlify.com/sites/jimnieters/deploys)

CHANGELOG.md
### 0.1.0 (2019-07-30)

##### Build System / Dependencies

* Fix the version of cms to fix the deploy ([bdbc90de](https://github.com/gatsbyjs/gatsby-starter-default/commit/bdbc90ded5b8e9a2e42da4c28553a66a0f517cbf))

##### Chores

* Update more content ([b9fd7d08](https://github.com/gatsbyjs/gatsby-starter-default/commit/b9fd7d088d157445743f62bb2fd6a3484640ded0))
* Remove test files ([4f120792](https://github.com/gatsbyjs/gatsby-starter-default/commit/4f120792b82dd1c66abe7cb94dab2fe999a9884c))

##### New Features

* Fixed Tags, added md support, cleanup ([2c9dde45](https://github.com/gatsbyjs/gatsby-starter-default/commit/2c9dde45e595300d8bc88125e28910fb2b4b4b4d))
* Testing addition of articles by files ([1903267b](https://github.com/gatsbyjs/gatsby-starter-default/commit/1903267bd22f85e27f99f2022c64560829aaeced))
* Small updates ([92ba3bd9](https://github.com/gatsbyjs/gatsby-starter-default/commit/92ba3bd916ffc61d3f2c6ec926f8b2daeb8a85f8))
* Small updates ([b1954242](https://github.com/gatsbyjs/gatsby-starter-default/commit/b195424276d2feb5368a8d40af79c813f5fdf49e))
* Add CMS editor for articles and tags ([3e95af3f](https://github.com/gatsbyjs/gatsby-starter-default/commit/3e95af3fd28085b15669c294eab50a4ba5cd900e))
* New Hero background, small clean up, fix responsive ([039395f4](https://github.com/gatsbyjs/gatsby-starter-default/commit/039395f49bc610f669ef55e5f65c53642f4a6b33))
* Add tag pages, minor fixes ([3385a643](https://github.com/gatsbyjs/gatsby-starter-default/commit/3385a6438f437f82f9a736b48a7ed2bff5a1cb9f))
* Add GA tracking ID ([f5b4ce89](https://github.com/gatsbyjs/gatsby-starter-default/commit/f5b4ce8980c7e8db55a2c5ce211c40feb1c2b88f))

##### Bug Fixes

* Small updates to CMS config and cleanup ([45b46da6](https://github.com/gatsbyjs/gatsby-starter-default/commit/45b46da6e0511739efc83e0ba8d762a404e8b761))
* Clean up old generate ([88b3a9e4](https://github.com/gatsbyjs/gatsby-starter-default/commit/88b3a9e479604ea0f8813eff8e15c08d42cfede9))
* Fix About page image issue ([f88c96eb](https://github.com/gatsbyjs/gatsby-starter-default/commit/f88c96eb88b391dde69783361ee98c41e2d4993e))
* Fix the GA ([1c301025](https://github.com/gatsbyjs/gatsby-starter-default/commit/1c3010254f308bd8af471ca234b26baa67b5deab))
* Change the default setting for anonymize in GA ([43199629](https://github.com/gatsbyjs/gatsby-starter-default/commit/43199629a652add34f3f3514d72ce9894c5d0d65))
* Update GA ID for analytics ([6a73bcb7](https://github.com/gatsbyjs/gatsby-starter-default/commit/6a73bcb7aed3e5e5dae44af7acb46f5563d398a1))
* Update font styles on hero, New tag page design ([9c06f069](https://github.com/gatsbyjs/gatsby-starter-default/commit/9c06f069bdcad9c1598c99be440a6ed285ec5e85))
* Change the subsequent heading font to sans-serif ([138c5d56](https://github.com/gatsbyjs/gatsby-starter-default/commit/138c5d5613dfa7f033b4b7dc221beb5a71f6600f))
* Fix content for article 7, Update few fonts ([4f77ec98](https://github.com/gatsbyjs/gatsby-starter-default/commit/4f77ec98037423f8c20b66213b6776364efcf3d7))

##### Other Changes

* //github.com/devanandb/jimnieters-online ([69845ba2](https://github.com/gatsbyjs/gatsby-starter-default/commit/69845ba2bc000555e8da087ef72088984c59e2e2))
* //github.com/devanandb/jimnieters-online ([f83fd6ff](https://github.com/gatsbyjs/gatsby-starter-default/commit/f83fd6ff54c8c846f075b2acf4827b03eb2b9820))
* //github.com/devanandb/jimnieters-online ([1628936f](https://github.com/gatsbyjs/gatsby-starter-default/commit/1628936fbc68d5debeed1a335ff0f438121326e5))

##### Refactors

* Major refactor to use markdown throughout ([ec3cfb8a](https://github.com/gatsbyjs/gatsby-starter-default/commit/ec3cfb8a74caed3a28fe563bf580c8aadef0c99c))
* Update cms config ([85ae465b](https://github.com/gatsbyjs/gatsby-starter-default/commit/85ae465b847adb88f9f3835bd29aaa71a56a4691))
* Update the article cms config ([68ce122a](https://github.com/gatsbyjs/gatsby-starter-default/commit/68ce122abd2fc6fea2c9ad84bcc1a5e71c885a7b))

#### 0.0.4 (2019-07-16)

##### Build System / Dependencies

* Add netlify caching for build optimize ([ffe5980c](https://github.com/gatsbyjs/gatsby-starter-default/commit/ffe5980cc14990eb41120f62cdd0c8bbe163e499))

##### New Features

* Add auto sitemap generator ([0d0d2f37](https://github.com/gatsbyjs/gatsby-starter-default/commit/0d0d2f377b673ea80c9034a0ed8c9a7a70abb37b))
* Added color-coded styles, responsive fixes, L2-addons ([330b9ba6](https://github.com/gatsbyjs/gatsby-starter-default/commit/330b9ba6f6c860d8a752613629dc7b4034ba46b9))
* Add scroll to top button ([0792a433](https://github.com/gatsbyjs/gatsby-starter-default/commit/0792a433d1ed89e32926efe766ce4da44bb4e8c8))
* Add anchor navigation to home, to take to l2 ([6649ceed](https://github.com/gatsbyjs/gatsby-starter-default/commit/6649ceed7339934f18604d6ef523b409c0c4f790))
* Refactor nav and add new case-studies tab ([ffd9d85c](https://github.com/gatsbyjs/gatsby-starter-default/commit/ffd9d85c8f15e154b7776fe442403f12bbe079c9))
* Add custom 404 ([5432d9c8](https://github.com/gatsbyjs/gatsby-starter-default/commit/5432d9c84b84a32b1453c6b89f64936193f6c4da))
* Add Crisp chat for online live chat ([93af2abe](https://github.com/gatsbyjs/gatsby-starter-default/commit/93af2abe0b5feaf113d3402a696cb2c29d914df5))
* Add wip about page ([61a4bb0b](https://github.com/gatsbyjs/gatsby-starter-default/commit/61a4bb0bb0f31517d8cf31ba8e62cef33f62ae6d))
* Add dark mode ([75198bb5](https://github.com/gatsbyjs/gatsby-starter-default/commit/75198bb523b7bc62ff68f34b125a0a385b58ba0b))
* Add SEO tags for category and detail page ([969cf6b4](https://github.com/gatsbyjs/gatsby-starter-default/commit/969cf6b474de1df3e54702aa7bbd68c09019b521))
* Huge refactor, Multiple article fixes ([7b28c9d0](https://github.com/gatsbyjs/gatsby-starter-default/commit/7b28c9d0584479f18c3c2c78f7af5f01080e7a64))

src/pages/index.js
import React from "react"
import queryString from 'query-string'
// import { Link } from "gatsby"

import { graphql } from "gatsby"
import Layout from "../components/layout"
import Header from "../components/header"

import HeroImage from "../components/hero-image"
import SEO from "../components/seo"
import ArticleCard from "../components/articlecard"
import { FaEnvelope, FaFacebookSquare, FaLinkedin, FaTwitterSquare, FaQuoteLeft  } from 'react-icons/fa';


class IndexPage extends React.Component {
	constructor(props) {
		super(props);
		let qparams = queryString.parse(this.props.location.search);
		this.state = {
			filter: qparams.category || 'all'
		};
		let category = qparams.category || 'all';
		this.state.articles = this.props.data.articles.nodes;
		if (qparams.category && qparams.category!=='all') {
			let filtered = this.props.data.articles.nodes.filter(function(article) {
				return article.frontmatter.category === category;
			});
			this.state.articles = filtered;
		} else {
			let filtered = this.props.data.articles.nodes;
			this.state.articles = filtered;
		}
		// console.log(this.state.articles);
		
		this.handleChange = this.handleChange.bind(this);
	}
	componentWillMount() {

	}
	handleChange(event) {
		let filter = event.target.value;
		this.setState({filter: filter});
		// let filtered =[];
		if (filter!=='all') {
			let filtered = this.props.data.articles.nodes.filter(function(article) {
				return article.frontmatter.category ===  filter;
			});
			this.setState({articles: filtered});
		} else {
			let filtered = this.props.data.articles.nodes;
			this.setState({articles: filtered});
		}
		
		
	}
	render() {
		const articles = this.state.articles;
		const socialLinks = [
			{
				"name": "linkedin",
				"url": "https://www.linkedin.com/in/jimnieters/en",
				"icon": "Linkedin"
			},
			{
				"name": "facebook",
				"url": "https://www.facebook.com/JimNieters",
				"icon": "Facebook"
			},
			{
				"name": "twitter",
				"url": "https://twitter.com/jimnieters",
				"icon": "Twitter"
			},
			{
				"name": "mail",
				"url": "mailto:jnieters@mac.com",
				"icon": "Envelope"
			}
		]
		return (
			<Layout className="dark:bg-gray-900">
				<SEO title="Home" />
				{/* <HeaderMinimal siteTitle="Jim Nieters - Work" /> */}
				<div className="hero-container sm:small bg-cover">
					<div className="container flex-wrap mx-auto pt-10 lg:my-0 px-8 py-4 lg:pt-32 sm:pt-20 relative">
						<h1 className="text-white dark:text-black font-extrabold font-serif mt-2 tracking-wider text-5xl sm:text-4xl lg:text-6xl">Jim Nieters</h1>
						<p className="mt-1 text-2xl lg:text-4xl font-medium tracking-normal text-gray-100 dark:text-black flex lg:justify-start">User Experience Design Leader</p>
						<p className="mt-1 text-gray-100 dark:text-gray-800 text-xl flex lg:justify-start font-normal lg:pb-10 tracking-normal">San Francisco, Bay Area</p>
						<div className="absolute opacity-25 -ml-6">
							<FaQuoteLeft className="text-6xl text-gray-500" />
						</div>
						<p className="mt-5 text-2xl md:text-2xl text-gray-100 dark:text-gray-300 w-full md:w-3/5 pb-10 font-light sm:pb-40 md:pb-20 lg:pb-16 summary">I am a ux leader who loves to inspire and be inspired by design. Great design disrupts markets and delights users.</p>
						
						<div className="flex flex-wrap w-1/5 md:w-1/2 mb-10">

							{/* {socialLinks.map()} */}
							{socialLinks.map((slink, i) => (
								<a className="mr-5 mb-8 text-gray-200 cursor-pointer shadow-md hover:shadow-lg text-5xl dark:text-gray-800" key={i} target={slink.name !=='mail' ? "_blank" :''} rel="noopener noreferrer" aria-label={slink.name} href={slink.url}>
									{(() => {
										switch (slink.name) {
											case 'linkedin':
												return <FaLinkedin className="" />
											case 'facebook':
												return <FaFacebookSquare className="" />
											case 'twitter':
												return <FaTwitterSquare className="" />
											case 'mail':
												return <FaEnvelope className="" />
											default :
												return null
										}
									})()}
									
								</a>
								// <div>{article.slug}</div>
							))}
						</div>
						
						<div className="w-full lg:w-2/5 relative hidden">
							{/* <img src="/images/jim-folio-3.jpg" alt="Jim Nieters" className="rounded-none w-full lg:rounded-lg shadow-2xl hidden lg:block" /> */}
							<HeroImage className="rounded-none w-full lg:rounded-lg shadow-2xl hidden lg:block" />
							<div className="absolute bottom-0 pb-5 inset-x-0 text-center">
								<a href="/about" className="uppercase text-white text-xl font-medium border border-white inline-block py-2 px-5 rounded-sm ">About me</a>
							</div>
						</div>
						
						{/* <div className="absolute top-0 right-0 h-12 w-18 p-4">
							<button className="js-change-theme focus:outline-none">🌙</button>
						</div> */}


					</div>
				</div>
				
				<Header className="mb-3" siteTitle="&nbsp;" />
				{/* <div className="dark:bg-gray-800 absolute w-full sticky top-0 z-20">
					<div className="py-4 container mx-auto">
						
						
					</div>
				</div> */}
				<div className="filter-nav hidden lg:inline-block pt-1 mx-auto text-left absolute sticky top-0 z-20">
					<form className="inline-block py-2 flex justify-center">
						<label className={`filter px-8 ${ this.state.filter === 'all' ? 'active bg-gray-700' : ''}`}>
							<input className="appearance-none" type="radio" value="all" checked={this.state.filter === 'all'} onChange={this.handleChange} />
							All
						</label>
						<label className={`filter px-8 ${ this.state.filter === 'leader' ? 'active bg-leader' : ''}`}>
							<input className="appearance-none" type="radio" value="leader" checked={this.state.filter === 'leader'} onChange={this.handleChange} />
							Leader
						</label>
						<label className={`filter px-8 ${ this.state.filter === 'designer' ? 'active bg-designer' : ''}`}>
							<input className="appearance-none" type="radio" value="designer" checked={this.state.filter === 'designer'} onChange={this.handleChange} />
							Designer
						</label>
					</form>
				</div>
				<div className="filter-nav-sm md:hidden pt-4 block mx-auto text-left">
					<form className="py-4 flex justify-center">
						<label className={`filter text-sm py-1 ${ this.state.filter === 'all' ? 'active bg-sp' : ''}`}>
							<input className="appearance-none" type="radio" value="all" checked={this.state.filter === 'all'} onChange={this.handleChange} />
							All
						</label>
						<label className={`filter text-sm py-1 ${ this.state.filter === 'leader' ? 'active bg-leader' : ''}`}>
							<input className="appearance-none" type="radio" value="leader" checked={this.state.filter === 'leader'} onChange={this.handleChange} />
							Leader
						</label>
						<label className={`filter text-sm py-1 ${ this.state.filter === 'designer' ? 'active bg-designer' : ''}`}>
							<input className="appearance-none" type="radio" value="designer" checked={this.state.filter === 'designer'} onChange={this.handleChange} />
							Designer
						</label>
					</form>
				</div>
				<div className="container pt-12 pb-12 mb-10 mx-auto">
					<div className="flex flex-wrap justify-left">
						{articles.map((article, i) => (
							<div key={i} className="w-full md:w-1/2 lg:w-1/3">
								<ArticleCard  {...article}>
								
								</ArticleCard>
							</div>
							// <div>{article.slug}</div>
						))}
					</div>
					
				</div>
			</Layout>
		)
	}
}


export default IndexPage

export const query = graphql`
	query {
		articles: allMarkdownRemark(sort: {fields: frontmatter___order, order: ASC}) {
			nodes {
				frontmatter {
					article_id
					title
					tagline
					slug
					summary
					category
					order
					created(fromNow: true)
					tags
					image {
						childImageSharp {
							fluid(maxWidth: 320) {
								...GatsbyImageSharpFluid
							}
						}
					}
				}
			} 
		}
	}
`;

